frontend/src/features/properties/list/PropertyListView.tsx

```diff
@@ -357,53 +406,115 @@ const PropertyListView: React.FC = () => {
+          <VerticalDivider />
+
+          <TooltipWrapper toolTipId="edit-financial-values" toolTip={'Edit financial values'}>
+            <EditIconButton>
+              <FaEdit data-testid="edit-icon" size={36} onClick={() => setEditable(!editable)} />
+            </EditIconButton>
+          </TooltipWrapper>
+          {editable && (
+            <TooltipWrapper
+              toolTipId="save-edited-financial-values"
+              toolTip={'Save financial values'}
+            >
+              <Button
+                data-testid="save-changes"
+                disabled={dirtyRows.length === 0}
+                onClick={() => {
+                  if (tableFormRef.current?.dirty) {
+                    tableFormRef.current.submitForm();
+                  }
+                }}
+              >
+                Save edits
+              </Button>
+            </TooltipWrapper>
+          )}
         </Container>
-        <Table<IProperty>
-          name="propertiesTable"
-          lockPageSize={true}
-          columns={columns}
-          data={data || []}
-          loading={data === undefined}
-          filterable
-          sort={sorting}
-          pageIndex={pageIndex}
-          onRequestData={handleRequestData}
-          pageCount={pageCount}
-          onSortChange={(column: string, direction: SortDirection) => {
-            if (!!direction) {
-              setSorting({ ...sorting, [column]: direction });
-            } else {
-              const data: any = { ...sorting };
-              delete data[column];
-              setSorting(data);
-            }
-          }}
-          canRowExpand={(val: any) => {
-            if (val.values.propertyTypeId === 0) {
-              return true;
-            } else {
-              return false;
-            }
-          }}
-          detailsPanel={{
-            render: val => {
-              if (expandData[val.id]) {
-                return <Buildings hideHeaders={true} data={expandData[val.id]} />;
+        <Formik
+          innerRef={tableFormRef as any}
+          initialValues={{ properties: data }}
+          enableReinitialize
+          onSubmit={async values => {
+            let nextProperties = [...values.properties!];
+            const changedRows = dirtyRows.map(change => {
+              const data = { ...values.properties![change.rowId] };
+
+              return { rowId: change.rowId, data } as any;
+            });
+            if (changedRows.length > 0) {
+              for (const change of changedRows) {
+                const apiProperty = toApiProperty(change.data as any);
+                const callApi = apiProperty.parcelId ? updateParcel : updateBuilding;
+                const response: any = await callApi(apiProperty.id, apiProperty);
+                nextProperties = nextProperties.map((item, index: number) => {
+                  if (index === change.rowId) {
+                    item = { ...item, rowVersion: response.rowVersion } as any;
```

Comment:
I wonder if it makes more sense to use the item from the response here. Thoughts?

Reply:
You are right, I will flatten the response object